### PR TITLE
Add bookmark_cleanup task

### PR DESCRIPTION
**Pull request by lealeelu**
refs #1421 

Adds the avalon:migrate:bookmark_cleanup task to run after all bookmarks have been moved over. It checks to make sure the MediaObject it references exists. If it doesn't exist, the Bookmark is destroyed.

### CODE REVIEW

File: lib/tasks/avalon.rake [hunk 1]
```diff
@@ -89,6 +89,23 @@ namespace :avalon do
         end
       end
     end
+
+    desc "Cleanup failed bookmarks"
+    task bookmark_cleanup: :environment do
+      Bookmark.all.each do |b|
+        if MediaObject.where(id: b.document_id).count > 0
```
Comment (cjcolvar) [inline lib/tasks/avalon.rake]: This seems a bit chatty.

File: lib/tasks/avalon.rake [hunk 2]
```diff
@@ -89,6 +89,23 @@ namespace :avalon do
...
+          puts "Bookmark pid: #{b.id} exists"
+        else
+          puts "Bookmark pid: #{b.id} does not exist. Deleting...."
```
Comment (cjcolvar) [inline lib/tasks/avalon.rake]: How about moving this to after the delete and say deleted instead of deleting?

File: lib/tasks/avalon.rake [hunk 3]
```diff
@@ -89,6 +89,23 @@ namespace :avalon do
...
+          begin
+            b.destroy
+          rescue Exception => e
+            puts "Failed to delete"
```
Comment (cjcolvar) [inline lib/tasks/avalon.rake]: Should these be puts or Rails.logger ?

Comment (lealeelu) [inline lib/tasks/avalon.rake]: Everything else was puts. I checked before deciding on puts. Now that we're getting to something that will be used in production, maybe we could change everything to logger?

File: lib/tasks/avalon.rake (hunk 2 above)
Comment (lealeelu) [inline lib/tasks/avalon.rake]: can do

File: lib/tasks/avalon.rake (hunk 1 above)
Comment (lealeelu) [inline lib/tasks/avalon.rake]: Do you mean the lines below this? I wanted to make sure it worked.

Comment (cjcolvar) [inline lib/tasks/avalon.rake]: I was meaning the existance message.

File: lib/tasks/avalon.rake (hunk 3 above)
Comment (cjcolvar) [inline lib/tasks/avalon.rake]: Hmmm...let me see what is recommended for this context (rake).

Comment (bkeese) [inline lib/tasks/avalon.rake]: I recommend output like this:

error message 1
error message 2
...
z bookmarks failed to deleted
y bookmarks deleted
x total bookmarks inspected

**Comment by cjcolvar** [comment]
:+1:  I'll wait for travis to merge

**Comment by cjcolvar** [comment]
Fixes #1421

**Pull request merged**